equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,431 @@
+	mappedOperations := make([]interface{}, len(operations))
+	for _, operation := range operations {
+		mappedOperation := make(map[string]interface{})
+		mappedOperation["equinix_status"] = string(*operation.EquinixStatus)
+		mappedOperations = append(mappedOperations, mappedOperation)
+	}
+
+	operationSet := schema.NewSet(
+		schema.HashResource(&schema.Resource{Schema: fabricNetworkOperationSch()}),
+		mappedOperations,
+	)
+	return operationSet
+}
+func simplifiedFabricNetworkChangeToTerra(networkChange *v4.SimplifiedNetworkChange) *schema.Set {
+	changes := []*v4.SimplifiedNetworkChange{networkChange}
+	mappedChanges := make([]interface{}, len(changes))
+	for _, change := range changes {
+		mappedChange := make(map[string]interface{})
+		mappedChange["href"] = change.Href
+		mappedChange["type"] = string(*change.Type_)
+		mappedChange["uuid"] = change.Uuid
+		mappedChanges = append(mappedChanges, mappedChange)
+	}
+
+	changeSet := schema.NewSet(
+		schema.HashResource(&schema.Resource{Schema: fabricNetworkChangeSch()}),
+		mappedChanges,
+	)
+	return changeSet
+}
+
+func setFabricNetworkMap(d *schema.ResourceData, nt v4.Network) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	err := equinix_schema.SetMap(d, map[string]interface{}{
+		"name":              nt.Name,
+		"href":              nt.Href,
+		"uuid":              nt.Uuid,
+		"type":              nt.Type_,
+		"scope":             nt.Scope,
+		"state":             nt.State,
+		"operation":         fabricNetworkOperationToTerra(nt.Operation),
+		"change":            simplifiedFabricNetworkChangeToTerra(nt.Change),
+		"location":          equinix_fabric_schema.LocationToTerra(nt.Location),
+		"notifications":     equinix_fabric_schema.NotificationsToTerra(nt.Notifications),
+		"project":           equinix_fabric_schema.ProjectToTerra(nt.Project),
+		"change_log":        equinix_fabric_schema.ChangeLogToTerra(nt.ChangeLog),
+		"connections_count": nt.ConnectionsCount,
+	})
+	if err != nil {
+		return diag.FromErr(equinix_errors.FormatFabricError(err))
+	}
+	return diags
+}
+func getFabricNetworkUpdateRequest(network v4.Network, d *schema.ResourceData) (v4.NetworkChangeOperation, error) {
+	changeOps := v4.NetworkChangeOperation{}
+	existingName := network.Name
+	updateNameVal := d.Get("name")
+
+	log.Printf("existing name %s, Update Name Request %s ", existingName, updateNameVal)
+
+	if existingName != updateNameVal {
+		changeOps = v4.NetworkChangeOperation{Op: "replace", Path: "/name", Value: &updateNameVal}
+	} else {
+		return changeOps, fmt.Errorf("nothing to update for the Fabric Network: %s", existingName)
+	}
+	return changeOps, nil
+}
+func resourceFabricNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	dbConn, err := waitUntilFabricNetworkIsProvisioned(d.Id(), meta, ctx)
```

Comment:
This method is already returning a formatted Fabric Error, so we don't need to do it again in the error check.